backend/app/api/routes/schedule.py

```diff
@@ -0,0 +1,57 @@
+import datetime
+import json
+
+from fastapi import APIRouter, HTTPException
+
+import app.services.schedule as schedule_service
+from app.api.deps import SessionDep
+from app.models import Schedule, ScheduleCreate, ScheduleData, SchedulePublic
+
+router = APIRouter()
+
+
+def verify_schedule(schedule: ScheduleData) -> None:
+    try:
+        datetime.datetime.strptime(
+            f"{schedule.startDate} {schedule.timesOfDay[0]}", "%Y-%m-%d %H:%M"
+        )
+        if hasattr(schedule, "endDate"):
+            datetime.datetime.strptime(
+                f"{schedule.endDate} {schedule.timesOfDay[0]}", "%Y-%m-%d %H:%M"
+            )
+
+    except ValueError:
+        raise HTTPException(
+            status_code=422,
+            detail="Schedule input is not valid.",
+        )
+
+
+def convert_db_schedule_to_json(db_schedule: Schedule | None) -> SchedulePublic | None:
+    if db_schedule is None:
+        return None
```

Comment:
Related to my comment below. Since you're doing the is None check in the path operation it makes sense to remove it here and change the union type of the function parameter to be Schedule. Not a big deal but seems redundant to say it can be Schedule or None and then return None if its None.